packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -81,116 +92,505 @@ const propTypes = {
    * The selected or entered date value to display in the date input.
    */
   value: PropTypes.string,
-  /**
-  * String that labels the current element. 'aria-label' must be present,
-  * for accessibility.
-  */
-  ariaLabel: PropTypes.string,
 };
 
 const defaultProps = {
+  ariaLabel: undefined,
   buttonRefCallback: undefined,
   inputAttributes: undefined,
   isIncomplete: false,
   isInvalid: false,
   name: undefined,
   onBlur: undefined,
+  onButtonFocus: undefined,
   onChange: undefined,
   onClick: undefined,
   onFocus: undefined,
-  onButtonFocus: undefined,
   onKeyDown: undefined,
   required: false,
   placeholder: undefined,
   value: undefined,
-  ariaLabel: undefined,
 };
 
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+const dateState = { day: '', month: '', year: '' };
+
+const DatePickerInput = (props) => {
+  const [date, dateDispatch] = useReducer(dateReducer, dateState);
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+
+  const setDayRef = useCallback(node => {
+    dayInputRef.current = node;
+  }, []);
+
+  const setMonthRef = useCallback(node => {
+    monthInputRef.current = node;
+  }, []);
+
+  const setYearRef = useCallback(node => {
+    yearInputRef.current = node;
+  }, []);
+
+  const {
+    ariaLabel,
+    buttonRefCallback,
+    inputAttributes,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onButtonFocus,
+    onChange,
+    onClick,
+    onFocus,
+    onKeyDown,
+    placeholder,
+    required,
+    value,
+    ...customProps
+  } = props;
+
+  const id = customProps.id ? customProps.id : undefined;
+  if (customProps.id) {
+    delete customProps.id;
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
```

Comment:
Can we have seperate method to delete this customprops content ? once the valus are being retrieved ?

Reply:
We could have a separate method, but i feel that in this case this is just a one time deletion on render and since the method will not be reused anywhere else, I think it would be ok to have it like this.

Reply:
What are we deleting these? I don't remember.

Reply:
Deconstructing these from `props` above would remove the need to delete the keys. `shouldShowPicker` is a private prop, but `onCalendarButtonClick` is not defined in the propTypes